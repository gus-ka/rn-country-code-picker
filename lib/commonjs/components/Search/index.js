"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _reactNative = require("react-native");
var _theme = require("../../theme");
var _icons = require("../icons");
var _styles = _interopRequireDefault(require("./styles"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Search = _ref => {
  let {
    value,
    onChangeText,
    onClearInput,
    inputStyle
  } = _ref;
  const [valueState, setValueState] = (0, _react.useState)(value);
  function handleChangeText(text) {
    onChangeText(text);
    setValueState(text);
  }
  function handleClear() {
    handleChangeText('');
    onClearInput();
  }
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: _styles.default.container
  }, /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: _styles.default.searchIconContainer
  }, /*#__PURE__*/_react.default.createElement(_icons.SvgSearch, {
    color: _theme.colors.grey
  })), /*#__PURE__*/_react.default.createElement(_reactNative.TextInput, {
    placeholder: "Country name or abbreviation",
    value: valueState,
    onChangeText: handleChangeText,
    style: [_styles.default.input, inputStyle],
    testID: "search-input"
  }), valueState ? /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    style: _styles.default.clearContainer
  }, /*#__PURE__*/_react.default.createElement(_reactNative.TouchableOpacity, {
    onPress: handleClear,
    testID: "clear-button"
  }, /*#__PURE__*/_react.default.createElement(_icons.SvgClose, {
    color: _theme.colors.white,
    width: 20,
    height: 20
  }))) : null);
};
var _default = Search;
exports.default = _default;
//# sourceMappingURL=index.js.map