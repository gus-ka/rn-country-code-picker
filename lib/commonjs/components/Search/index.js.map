{"version":3,"names":["_react","_interopRequireWildcard","require","_reactNative","_theme","_icons","_styles","_interopRequireDefault","obj","__esModule","default","_getRequireWildcardCache","nodeInterop","WeakMap","cacheBabelInterop","cacheNodeInterop","cache","has","get","newObj","hasPropertyDescriptor","Object","defineProperty","getOwnPropertyDescriptor","key","prototype","hasOwnProperty","call","desc","set","Search","_ref","value","onChangeText","onClearInput","inputStyle","valueState","setValueState","useState","handleChangeText","text","handleClear","createElement","View","style","styles","container","searchIconContainer","SvgSearch","color","colors","grey","TextInput","placeholder","input","testID","clearContainer","TouchableOpacity","onPress","SvgClose","white","width","height","_default","exports"],"sources":["index.tsx"],"sourcesContent":["import React, { useEffect, useState } from 'react';\nimport { ScrollView, TextInput, TouchableOpacity, View } from 'react-native';\nimport { colors } from '../../theme';\nimport { ISearchProps } from '../../types';\nimport { SvgClose, SvgSearch } from '../icons';\nimport styles from './styles';\n\nconst Search: React.FC<ISearchProps> = ({ value, onChangeText, onClearInput, inputStyle }) => {\n  const [valueState, setValueState] = useState(value);\n\n  function handleChangeText(text: string) {\n    onChangeText(text);\n    setValueState(text);\n  }\n\n  function handleClear() {\n    handleChangeText('');\n    onClearInput();\n  }\n\n  return (\n    <View style={styles.container}>\n      <View style={styles.searchIconContainer}>\n        <SvgSearch color={colors.grey} />\n      </View>\n      <TextInput\n        placeholder=\"Country name or abbreviation\"\n        value={valueState}\n        onChangeText={handleChangeText}\n        style={[styles.input, inputStyle]}\n        testID=\"search-input\"\n      />\n      {valueState ? (\n        <View style={styles.clearContainer}>\n          <TouchableOpacity onPress={handleClear} testID=\"clear-button\">\n            <SvgClose color={colors.white} width={20} height={20} />\n          </TouchableOpacity>\n        </View>\n      ) : null}\n    </View>\n  );\n};\n\nexport default Search;\n"],"mappings":";;;;;;AAAA,IAAAA,MAAA,GAAAC,uBAAA,CAAAC,OAAA;AACA,IAAAC,YAAA,GAAAD,OAAA;AACA,IAAAE,MAAA,GAAAF,OAAA;AAEA,IAAAG,MAAA,GAAAH,OAAA;AACA,IAAAI,OAAA,GAAAC,sBAAA,CAAAL,OAAA;AAA8B,SAAAK,uBAAAC,GAAA,WAAAA,GAAA,IAAAA,GAAA,CAAAC,UAAA,GAAAD,GAAA,KAAAE,OAAA,EAAAF,GAAA;AAAA,SAAAG,yBAAAC,WAAA,eAAAC,OAAA,kCAAAC,iBAAA,OAAAD,OAAA,QAAAE,gBAAA,OAAAF,OAAA,YAAAF,wBAAA,YAAAA,CAAAC,WAAA,WAAAA,WAAA,GAAAG,gBAAA,GAAAD,iBAAA,KAAAF,WAAA;AAAA,SAAAX,wBAAAO,GAAA,EAAAI,WAAA,SAAAA,WAAA,IAAAJ,GAAA,IAAAA,GAAA,CAAAC,UAAA,WAAAD,GAAA,QAAAA,GAAA,oBAAAA,GAAA,wBAAAA,GAAA,4BAAAE,OAAA,EAAAF,GAAA,UAAAQ,KAAA,GAAAL,wBAAA,CAAAC,WAAA,OAAAI,KAAA,IAAAA,KAAA,CAAAC,GAAA,CAAAT,GAAA,YAAAQ,KAAA,CAAAE,GAAA,CAAAV,GAAA,SAAAW,MAAA,WAAAC,qBAAA,GAAAC,MAAA,CAAAC,cAAA,IAAAD,MAAA,CAAAE,wBAAA,WAAAC,GAAA,IAAAhB,GAAA,QAAAgB,GAAA,kBAAAH,MAAA,CAAAI,SAAA,CAAAC,cAAA,CAAAC,IAAA,CAAAnB,GAAA,EAAAgB,GAAA,SAAAI,IAAA,GAAAR,qBAAA,GAAAC,MAAA,CAAAE,wBAAA,CAAAf,GAAA,EAAAgB,GAAA,cAAAI,IAAA,KAAAA,IAAA,CAAAV,GAAA,IAAAU,IAAA,CAAAC,GAAA,KAAAR,MAAA,CAAAC,cAAA,CAAAH,MAAA,EAAAK,GAAA,EAAAI,IAAA,YAAAT,MAAA,CAAAK,GAAA,IAAAhB,GAAA,CAAAgB,GAAA,SAAAL,MAAA,CAAAT,OAAA,GAAAF,GAAA,MAAAQ,KAAA,IAAAA,KAAA,CAAAa,GAAA,CAAArB,GAAA,EAAAW,MAAA,YAAAA,MAAA;AAE9B,MAAMW,MAA8B,GAAGC,IAAA,IAAuD;EAAA,IAAtD;IAAEC,KAAK;IAAEC,YAAY;IAAEC,YAAY;IAAEC;EAAW,CAAC,GAAAJ,IAAA;EACvF,MAAM,CAACK,UAAU,EAAEC,aAAa,CAAC,GAAG,IAAAC,eAAQ,EAACN,KAAK,CAAC;EAEnD,SAASO,gBAAgBA,CAACC,IAAY,EAAE;IACtCP,YAAY,CAACO,IAAI,CAAC;IAClBH,aAAa,CAACG,IAAI,CAAC;EACrB;EAEA,SAASC,WAAWA,CAAA,EAAG;IACrBF,gBAAgB,CAAC,EAAE,CAAC;IACpBL,YAAY,CAAC,CAAC;EAChB;EAEA,oBACElC,MAAA,CAAAU,OAAA,CAAAgC,aAAA,CAACvC,YAAA,CAAAwC,IAAI;IAACC,KAAK,EAAEC,eAAM,CAACC;EAAU,gBAC5B9C,MAAA,CAAAU,OAAA,CAAAgC,aAAA,CAACvC,YAAA,CAAAwC,IAAI;IAACC,KAAK,EAAEC,eAAM,CAACE;EAAoB,gBACtC/C,MAAA,CAAAU,OAAA,CAAAgC,aAAA,CAACrC,MAAA,CAAA2C,SAAS;IAACC,KAAK,EAAEC,aAAM,CAACC;EAAK,CAAE,CAC5B,CAAC,eACPnD,MAAA,CAAAU,OAAA,CAAAgC,aAAA,CAACvC,YAAA,CAAAiD,SAAS;IACRC,WAAW,EAAC,8BAA8B;IAC1CrB,KAAK,EAAEI,UAAW;IAClBH,YAAY,EAAEM,gBAAiB;IAC/BK,KAAK,EAAE,CAACC,eAAM,CAACS,KAAK,EAAEnB,UAAU,CAAE;IAClCoB,MAAM,EAAC;EAAc,CACtB,CAAC,EACDnB,UAAU,gBACTpC,MAAA,CAAAU,OAAA,CAAAgC,aAAA,CAACvC,YAAA,CAAAwC,IAAI;IAACC,KAAK,EAAEC,eAAM,CAACW;EAAe,gBACjCxD,MAAA,CAAAU,OAAA,CAAAgC,aAAA,CAACvC,YAAA,CAAAsD,gBAAgB;IAACC,OAAO,EAAEjB,WAAY;IAACc,MAAM,EAAC;EAAc,gBAC3DvD,MAAA,CAAAU,OAAA,CAAAgC,aAAA,CAACrC,MAAA,CAAAsD,QAAQ;IAACV,KAAK,EAAEC,aAAM,CAACU,KAAM;IAACC,KAAK,EAAE,EAAG;IAACC,MAAM,EAAE;EAAG,CAAE,CACvC,CACd,CAAC,GACL,IACA,CAAC;AAEX,CAAC;AAAC,IAAAC,QAAA,GAEajC,MAAM;AAAAkC,OAAA,CAAAtD,OAAA,GAAAqD,QAAA"}