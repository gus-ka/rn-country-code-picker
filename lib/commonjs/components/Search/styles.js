"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _reactNative = require("react-native");
var _theme = require("../../theme");
var _default = _reactNative.StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'space-between',
    alignSelf: 'center',
    flexDirection: 'row'
  },
  searchIconContainer: {
    alignSelf: 'center'
  },
  input: {
    marginHorizontal: _theme.spacing.s,
    marginVertical: _theme.spacing.s,
    fontSize: 17,
    flex: 1
  },
  clearContainer: {
    backgroundColor: _theme.colors.greyLight,
    height: 20,
    width: 20,
    borderRadius: 10,
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center'
  }
});
exports.default = _default;
//# sourceMappingURL=styles.js.map