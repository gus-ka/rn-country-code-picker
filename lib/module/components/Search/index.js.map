{"version":3,"names":["React","useState","TextInput","TouchableOpacity","View","colors","SvgClose","SvgSearch","styles","Search","_ref","value","onChangeText","onClearInput","inputStyle","valueState","setValueState","handleChangeText","text","handleClear","createElement","style","container","searchIconContainer","color","grey","placeholder","input","testID","clearContainer","onPress","white","width","height"],"sources":["index.tsx"],"sourcesContent":["import React, { useEffect, useState } from 'react';\nimport { ScrollView, TextInput, TouchableOpacity, View } from 'react-native';\nimport { colors } from '../../theme';\nimport { ISearchProps } from '../../types';\nimport { SvgClose, SvgSearch } from '../icons';\nimport styles from './styles';\n\nconst Search: React.FC<ISearchProps> = ({ value, onChangeText, onClearInput, inputStyle }) => {\n  const [valueState, setValueState] = useState(value);\n\n  function handleChangeText(text: string) {\n    onChangeText(text);\n    setValueState(text);\n  }\n\n  function handleClear() {\n    handleChangeText('');\n    onClearInput();\n  }\n\n  return (\n    <View style={styles.container}>\n      <View style={styles.searchIconContainer}>\n        <SvgSearch color={colors.grey} />\n      </View>\n      <TextInput\n        placeholder=\"Country name or abbreviation\"\n        value={valueState}\n        onChangeText={handleChangeText}\n        style={[styles.input, inputStyle]}\n        testID=\"search-input\"\n      />\n      {valueState ? (\n        <View style={styles.clearContainer}>\n          <TouchableOpacity onPress={handleClear} testID=\"clear-button\">\n            <SvgClose color={colors.white} width={20} height={20} />\n          </TouchableOpacity>\n        </View>\n      ) : null}\n    </View>\n  );\n};\n\nexport default Search;\n"],"mappings":"AAAA,OAAOA,KAAK,IAAeC,QAAQ,QAAQ,OAAO;AAClD,SAAqBC,SAAS,EAAEC,gBAAgB,EAAEC,IAAI,QAAQ,cAAc;AAC5E,SAASC,MAAM,QAAQ,aAAa;AAEpC,SAASC,QAAQ,EAAEC,SAAS,QAAQ,UAAU;AAC9C,OAAOC,MAAM,MAAM,UAAU;AAE7B,MAAMC,MAA8B,GAAGC,IAAA,IAAuD;EAAA,IAAtD;IAAEC,KAAK;IAAEC,YAAY;IAAEC,YAAY;IAAEC;EAAW,CAAC,GAAAJ,IAAA;EACvF,MAAM,CAACK,UAAU,EAAEC,aAAa,CAAC,GAAGf,QAAQ,CAACU,KAAK,CAAC;EAEnD,SAASM,gBAAgBA,CAACC,IAAY,EAAE;IACtCN,YAAY,CAACM,IAAI,CAAC;IAClBF,aAAa,CAACE,IAAI,CAAC;EACrB;EAEA,SAASC,WAAWA,CAAA,EAAG;IACrBF,gBAAgB,CAAC,EAAE,CAAC;IACpBJ,YAAY,CAAC,CAAC;EAChB;EAEA,oBACEb,KAAA,CAAAoB,aAAA,CAAChB,IAAI;IAACiB,KAAK,EAAEb,MAAM,CAACc;EAAU,gBAC5BtB,KAAA,CAAAoB,aAAA,CAAChB,IAAI;IAACiB,KAAK,EAAEb,MAAM,CAACe;EAAoB,gBACtCvB,KAAA,CAAAoB,aAAA,CAACb,SAAS;IAACiB,KAAK,EAAEnB,MAAM,CAACoB;EAAK,CAAE,CAC5B,CAAC,eACPzB,KAAA,CAAAoB,aAAA,CAAClB,SAAS;IACRwB,WAAW,EAAC,8BAA8B;IAC1Cf,KAAK,EAAEI,UAAW;IAClBH,YAAY,EAAEK,gBAAiB;IAC/BI,KAAK,EAAE,CAACb,MAAM,CAACmB,KAAK,EAAEb,UAAU,CAAE;IAClCc,MAAM,EAAC;EAAc,CACtB,CAAC,EACDb,UAAU,gBACTf,KAAA,CAAAoB,aAAA,CAAChB,IAAI;IAACiB,KAAK,EAAEb,MAAM,CAACqB;EAAe,gBACjC7B,KAAA,CAAAoB,aAAA,CAACjB,gBAAgB;IAAC2B,OAAO,EAAEX,WAAY;IAACS,MAAM,EAAC;EAAc,gBAC3D5B,KAAA,CAAAoB,aAAA,CAACd,QAAQ;IAACkB,KAAK,EAAEnB,MAAM,CAAC0B,KAAM;IAACC,KAAK,EAAE,EAAG;IAACC,MAAM,EAAE;EAAG,CAAE,CACvC,CACd,CAAC,GACL,IACA,CAAC;AAEX,CAAC;AAED,eAAexB,MAAM"}