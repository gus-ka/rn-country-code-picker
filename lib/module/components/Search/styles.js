import { StyleSheet } from 'react-native';
import { colors, spacing } from '../../theme';
export default StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'space-between',
    alignSelf: 'center',
    flexDirection: 'row'
  },
  searchIconContainer: {
    alignSelf: 'center'
  },
  input: {
    marginHorizontal: spacing.s,
    marginVertical: spacing.s,
    fontSize: 17,
    flex: 1
  },
  clearContainer: {
    backgroundColor: colors.greyLight,
    height: 20,
    width: 20,
    borderRadius: 10,
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center'
  }
});
//# sourceMappingURL=styles.js.map