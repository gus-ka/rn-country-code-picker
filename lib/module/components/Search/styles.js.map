{"version":3,"names":["StyleSheet","colors","spacing","create","container","width","justifyContent","alignSelf","flexDirection","searchIconContainer","input","marginHorizontal","s","marginVertical","fontSize","flex","clearContainer","backgroundColor","greyLight","height","borderRadius","alignItems"],"sources":["styles.ts"],"sourcesContent":["import { StyleSheet } from 'react-native';\nimport { colors, spacing } from '../../theme';\n\nexport default StyleSheet.create({\n  container: {\n    width: '100%',\n    justifyContent: 'space-between',\n    alignSelf: 'center',\n    flexDirection: 'row',\n  },\n  searchIconContainer: {\n    alignSelf: 'center',\n  },\n  input: {\n    marginHorizontal: spacing.s,\n    marginVertical: spacing.s,\n    fontSize: 17,\n    flex: 1,\n  },\n  clearContainer: {\n    backgroundColor: colors.greyLight,\n    height: 20,\n    width: 20,\n    borderRadius: 10,\n    alignSelf: 'center',\n    justifyContent: 'center',\n    alignItems: 'center',\n  },\n});\n"],"mappings":"AAAA,SAASA,UAAU,QAAQ,cAAc;AACzC,SAASC,MAAM,EAAEC,OAAO,QAAQ,aAAa;AAE7C,eAAeF,UAAU,CAACG,MAAM,CAAC;EAC/BC,SAAS,EAAE;IACTC,KAAK,EAAE,MAAM;IACbC,cAAc,EAAE,eAAe;IAC/BC,SAAS,EAAE,QAAQ;IACnBC,aAAa,EAAE;EACjB,CAAC;EACDC,mBAAmB,EAAE;IACnBF,SAAS,EAAE;EACb,CAAC;EACDG,KAAK,EAAE;IACLC,gBAAgB,EAAET,OAAO,CAACU,CAAC;IAC3BC,cAAc,EAAEX,OAAO,CAACU,CAAC;IACzBE,QAAQ,EAAE,EAAE;IACZC,IAAI,EAAE;EACR,CAAC;EACDC,cAAc,EAAE;IACdC,eAAe,EAAEhB,MAAM,CAACiB,SAAS;IACjCC,MAAM,EAAE,EAAE;IACVd,KAAK,EAAE,EAAE;IACTe,YAAY,EAAE,EAAE;IAChBb,SAAS,EAAE,QAAQ;IACnBD,cAAc,EAAE,QAAQ;IACxBe,UAAU,EAAE;EACd;AACF,CAAC,CAAC"}