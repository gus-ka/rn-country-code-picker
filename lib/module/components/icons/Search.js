function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import Svg, { Path } from 'react-native-svg';
function SvgSearch(props) {
  return /*#__PURE__*/React.createElement(Svg, _extends({
    width: 24,
    height: 24,
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24"
  }, props), /*#__PURE__*/React.createElement(Path, {
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: 2,
    d: "M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
  }));
}
export default SvgSearch;
//# sourceMappingURL=Search.js.map