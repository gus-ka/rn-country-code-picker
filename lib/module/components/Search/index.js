import React, { useState } from 'react';
import { TextInput, TouchableOpacity, View } from 'react-native';
import { colors } from '../../theme';
import { SvgClose, SvgSearch } from '../icons';
import styles from './styles';
const Search = _ref => {
  let {
    value,
    onChangeText,
    onClearInput,
    inputStyle
  } = _ref;
  const [valueState, setValueState] = useState(value);
  function handleChangeText(text) {
    onChangeText(text);
    setValueState(text);
  }
  function handleClear() {
    handleChangeText('');
    onClearInput();
  }
  return /*#__PURE__*/React.createElement(View, {
    style: styles.container
  }, /*#__PURE__*/React.createElement(View, {
    style: styles.searchIconContainer
  }, /*#__PURE__*/React.createElement(SvgSearch, {
    color: colors.grey
  })), /*#__PURE__*/React.createElement(TextInput, {
    placeholder: "Country name or abbreviation",
    value: valueState,
    onChangeText: handleChangeText,
    style: [styles.input, inputStyle],
    testID: "search-input"
  }), valueState ? /*#__PURE__*/React.createElement(View, {
    style: styles.clearContainer
  }, /*#__PURE__*/React.createElement(TouchableOpacity, {
    onPress: handleClear,
    testID: "clear-button"
  }, /*#__PURE__*/React.createElement(SvgClose, {
    color: colors.white,
    width: 20,
    height: 20
  }))) : null);
};
export default Search;
//# sourceMappingURL=index.js.map