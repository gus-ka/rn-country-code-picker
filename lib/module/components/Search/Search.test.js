import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import Search from './';
const onChangeTextMockFn = jest.fn();
const onClearInputMockFn = jest.fn();
const SEARCH_VALUE = 'Turkey';
const SEARCH_COMPONENT = /*#__PURE__*/React.createElement(Search, {
  onChangeText: onChangeTextMockFn,
  onClearInput: onClearInputMockFn
});
test('renders correctly', () => {
  const {
    toJSON
  } = render(SEARCH_COMPONENT);
  const rendered = toJSON();
  expect(rendered).toMatchSnapshot();
});
test('changes the input value', () => {
  const {
    getByTestId
  } = render(SEARCH_COMPONENT);
  const textInput = getByTestId('search-input');
  fireEvent.changeText(textInput, SEARCH_VALUE);
  expect(onChangeTextMockFn).toBeCalledWith(SEARCH_VALUE);
  expect(textInput.props.value).toBe(SEARCH_VALUE);
});
test('clears the input value', () => {
  const {
    getByTestId
  } = render(SEARCH_COMPONENT);
  const textInput = getByTestId('search-input');
  fireEvent.changeText(textInput, SEARCH_VALUE);
  expect(onChangeTextMockFn).toBeCalledWith(SEARCH_VALUE);
  expect(textInput.props.value).toBe(SEARCH_VALUE);
  const clearButton = getByTestId('clear-button');
  fireEvent.press(clearButton);
  expect(onClearInputMockFn).toBeCalled();
  expect(textInput.props.value).toBe('');
  expect(onChangeTextMockFn).toBeCalledWith('');
});
//# sourceMappingURL=Search.test.js.map