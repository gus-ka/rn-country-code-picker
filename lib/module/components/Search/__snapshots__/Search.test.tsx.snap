// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={24}
      bbWidth={24}
      color={4291085508}
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="currentColor"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      tintColor={4291085508}
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
          ]
        }
        stroke={
          Array [
            2,
          ]
        }
      >
        <RNSVGPath
          d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
          propList={
            Array [
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <TextInput
    onChangeText={[Function]}
    placeholder="Country name or abbreviation"
    style={
      Array [
        Object {
          "flex": 1,
          "fontSize": 17,
          "marginHorizontal": 8,
          "marginVertical": 8,
        },
        undefined,
      ]
    }
    testID="search-input"
  />
</View>
`;
